Why use MSFT namespace for DI extensions? (#20549)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/dependency-injection.md

[!code-csharp[](dependency-injection/samples/2.x/DependencyInjectionSample/Startup.cs?name=snippet1&highlight=5)]

> [!NOTE]

> > Each `services.Add{SERVICE_NAME}` extension method adds (and potentially configures) services. For example, `services.AddMvc()` adds the services Razor Pages and MVC require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations. Using the namespace `Microsoft.Extensions.DependencyInjection` for DI extension methods allows them to be displayed in Intellisense.

Reviewer: ```suggestion
> Each `services.Add{SERVICE_NAME}` extension method adds (and potentially configures) services. For example, `services.AddMvc()` adds the services Razor Pages and MVC require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations. Using the namespace `Microsoft.Extensions.DependencyInjection` for DI extension methods allows them to be displayed in Intellisense without any additional `using` blocks added by the developer. Also, it keeps the `Startup` class (where these extension methods are mostly called from) clean from too many `using` blocks.
```

> > Each `services.Add{SERVICE_NAME}` extension method adds, and potentially configures, services. For example, `services.AddMvc()` adds the services Razor Pages and MVC require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations. Using the namespace `Microsoft.Extensions.DependencyInjection` for DI extension methods also allows them to be displayed in Intellisense.

Reviewer: ```suggestion
> Each `services.Add{SERVICE_NAME}` extension method adds, and potentially configures, services. For example, `services.AddMvc()` adds the services Razor Pages and MVC require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations. Using the namespace `Microsoft.Extensions.DependencyInjection` for DI extension methods also allows them to be displayed in Intellisense without any additional `using` blocks added by the developer.
 Also, it keeps the `Startup` class (where these extension methods are mostly called from) clean from too many `using` blocks.
```
```

[!code-csharp[](dependency-injection/samples/2.x/DependencyInjectionSample/Startup.cs?name=snippet1&highlight=5)]

> [!NOTE]

> > Each `services.Add{SERVICE_NAME}` extension method adds, and potentially configures, services. For example, `services.AddControllersWithViews`, `services.AddRazorPages`, and `services.AddControllers` adds the services ASP.NET Core apps require. We recommended that apps follow this convention. Place extension methods in the [Microsoft.Extensions.DependencyInjection](/dotnet/api/microsoft.extensions.dependencyinjection) namespace to encapsulate groups of service registrations. Including the namespace portion `Microsoft.Extensions.DependencyInjection` for DI extension methods also:

Reviewer 2: ```suggestion
> Each `services.Add{SERVICE_NAME}` extension method adds, and potentially configures, services. For example, `services.AddControllersWithViews`, `services.AddRazorPages`, and `services.AddControllers` adds the services ASP.NET Core apps require. We recommended that apps follow this convention. Place extension methods in the <xref:Microsoft.Extensions.DependencyInjection?displayProperty=fullName> namespace to encapsulate groups of service registrations. Including the namespace portion `Microsoft.Extensions.DependencyInjection` for DI extension methods also:
>
```

> > * Allows them to be displayed in Intellisense with out adding additional `using` blocks.

Reviewer 2: ```suggestion
> * Allows them to be displayed in [IntelliSense](/visualstudio/ide/using-intellisense) with out adding additional `using` blocks.
```
